model.py

```diff
@@ -154,41 +195,41 @@ def forward(self, latent_image: Tensor, input_ids: Tensor, timestep: Tensor, tex
         # NOTE: In order to undo the patching, we have a linear layer back to NUM_PATCHES and shuffle data to reform an image with same dim
         # NOTE: Recover the latent image channels from hidden dimension
         # Pluck out last NP tokens correspoding to patch embeddings
-        projected_img_emb = self.transformer_img_proj(transformer_emb[:, -NUM_PATCHES:, :])  # (S[B, :-NP, C]) @ (C, LC*L*L/NP) -> (B, NP, LC*L*L/NP)
+        projected_img_emb = self.transformer_img_proj(transformer_emb[:, -self.num_patches:, :])  # (S[B, :-NP, C]) @ (C, LC*L*L/NP) -> (B, NP, LC*L*L/NP)
         # print("projected_img_emb after projection:", projected_img_emb.shape)
         
         # Pure reshaping to original image shape, no learnable parameters
         # (B, NP, LC*L*L/NP) -> (B, sqrt[NP], sqrt[NP], L/sqrt[NP], L/sqrt[NP], LC) | B*LC*L^2 on both sides
         # NOTE: NP = (LATENT_DIM // PATCH_DIM)^2
-        projected_img_emb = projected_img_emb.reshape(
-            shape=(B, LATENT_DIM // PATCH_DIM, LATENT_DIM // PATCH_DIM, PATCH_DIM, PATCH_DIM, LATENT_CHANNELS)
+        # projected_img_emb = projected_img_emb.reshape(
+        #     shape=(B, self.latent_dim // self.patch_dim, self.latent_dim // self.patch_dim, self.patch_dim, self.patch_dim, self.vae_latent_channels)
+        # )
+        denoised_latent_image = rearrange(
```

Comment:
Can you adjust the shaping comments accordingly? Maybe just have all three reshapes in a row or perhaps explain a bit more about what's going on here.